File: lc_soln/0209_minimum_size_subarray_sum/solution.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n = nums.size();
        int result = n + 1;
        int curr = 0;
        int start = 0;
        for (int i = 0; i < n; ++i) {
            curr += nums[i];
            while (curr >= target) {
                result = min(result, i - start + 1);
                curr -= nums[start];
                ++start;
            }
        }
        return result == n + 1 ? 0 : result;
    }
};

File: lc_soln/0209_minimum_size_subarray_sum/solution.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        start = 0
        curr = 0
        n = len(nums)
        result = n + 1
        for i, num in enumerate(nums):
            curr += num
            while curr >= target:
                result = min(result, i - start + 1)
                curr -= nums[start]
                start += 1
        return 0 if result == n + 1 else result
